consutil/main.py

```diff
@@ -62,10 +67,56 @@ def clear(linenum):
 
 # 'connect' subcommand
 @consutil.command()
-@click.argument('linenum')
-def connect(linenum):
-    """Connect to switch via console device"""
-    click.echo("connect linenum")
+@click.argument('target')
+@click.option('--devicename', '-d', is_flag=True, help="connect by name - if flag is set, interpret linenum as device name instead")
+def connect(target, devicename):
+    """Connect to switch via console device - TARGET is line number or device name of switch"""
+    lineNumber = ""
+    if devicename:
+        lineNumber = getLineNumber(target)
+        if lineNumber == "":
+            click.echo("Device {} does not exist".format(target))
+            return
+    else:
+        lineNumber = target
+    checkDevice(lineNumber)
+    lineNumber = str(lineNumber)
+
```

Comment:
Code below this comment can probably go to a helper function like connecToLine or similar. So that we don't have a long function. Also it makes easier if we decide to have 2 command for connecting to line/device.

Reply:
How about moving the code above the comment into getLineNumber()? So that all the connect logic is in connect(), and all the get line logic is in getLineNumber()